plugin/core/views.py

```diff
@@ -172,3 +178,42 @@ def text_document_range_formatting(view: sublime.View, region: sublime.Region) -
         "options": formatting_options(view.settings()),
         "range": region_to_range(view, region).to_lsp()
     })
+
+
+def minihtml(view: sublime.View, content: Union[str, dict], popup_width: Optional[int] = None) -> str:
+    """ Content can be a string or MarkupContent. `popup_width` will wrap text based on the provided width. """
+    if isinstance(content, str):
+        return text2html(view, content, popup_width)
+    elif isinstance(content, dict):
+        value = content.get("value") or ""
+        kind = content.get("kind")
+        if kind == "markdown":
+            return mdpopups.md2html(view, value)
+        else:
+            # must be plaintext
+            return text2html(view, value, popup_width)
+    else:
+        return ''
+
+
+def text2html(view: sublime.View, content: str, popup_width: Optional[int] = None) -> str:
```

Comment:
Here is my suggestion:
```py

RE = re.compile(r'(?P<entity>[&<>])|(?P<url>https?://(?:[\w\d:#@%/;$()~_?\+\-=\\\.&](?:#!)?)*)', flags=re.IGNORECASE)
ENTITY_MAP = {
    "&": "&amp;",
    "<": "&lt;",
    ">": "&gt;",
}


def text2html(view: sublime.View, content: str, popup_width: Optional[int] = None) -> str:
    # if popup_width is not None:
    #     # if popup width is provided, calucalte how many caracters can fit on one line
    #     wrap_width = int(popup_width / view.em_width())
    #     content = "<br>".join(textwrap.wrap(content, width=wrap_width,
    #                           replace_whitespace=False, break_long_words=False, break_on_hyphens=False))

    def replace_match(match: Any) -> str:
        entity_match = match.group('entity')
        if entity_match:
            return ENTITY_MAP[entity_match]
        url = match.group('url')
        return "<a href='{}'>{}</a>".format(url, url)

    content = re.sub(RE, replace_match, content).replace('\n', '<br>').replace('\t', '&nbsp;&nbsp;&nbsp;&nbsp;')

    # if there are 2 or more spaces, replace them with &nbsp;
    return re.sub(r"( {2,}|[\xc2\xa0])", lambda match: "&nbsp;" * len(match.group(0)), content)
```

- fixes link wrapping
- also replaces non-breaking spaces which helps with alignment in those "tables" I've mentioned before
- disabled text wrapping as I think it will do more harm than good and also it breaks formatting if we add '<be>s' before replacing.

Reply:
It could be massaged a bit more to have only one replace function instead of two separate ones.

Reply:
@rchl can you test what will happen if the content was
```
# I added " quotes which will be escaped to &qout; or something like that
content = 'Extensible JSON "http://json.org" encoder'
```
I think that the result will be `...://json.org&qout;</a>` or maybe not. 
But I have a feeling that it will

Reply:
I'm not escaping quotes in my code.
The `html.escape()` does it by default (but you can disable it with an argument). I don't think we need to escape quotes but it's easy to do that if we want.

Reply:
![Screenshot 2020-05-12 at 22 54 38](https://user-images.githubusercontent.com/153197/81744483-934b7f00-94a3-11ea-8eb6-7b499504583b.png)

Reply:
What if a url actually contains `&gt;`, `http://json.org&gt;`. Did we broke the link than?

Reply:
That's an invalid URL in the first place. Where does that example come from?

We can't fix incorrect code ourselves. Or we shouldn't need to at least. That would go a bit too far.